### PR TITLE
Delete many deployment e2e tests.

### PR DESCRIPTION
**Pull request by spxtr**
See #40839 for context. I left a few tests, but feel free to suggest more if there isn't adequate unit test coverage.
**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by janetkuo** [comment]
@spxtr I understand why you'd want to remove them, but I oppose removing these deployment e2e tests. Currently, unit tests only cover things like "what informer actions are expected to be triggered after we sync a Deployment once". Deployment controller can't be fully covered by those unit tests. What's more, we've seen flakes in e2e tests turned out to be real issues in deployments. Without those e2e tests we may end up having unstable and buggy deployment controller. 

@kargakis WDYT?

**Comment by spxtr** [comment]
> Currently, unit tests only cover things like "what informer actions are expected to be triggered after we sync a Deployment once". Deployment controller can't be fully covered by those unit tests.

Can you say a little more about why that is?

**Comment by janetkuo** [comment]
For example, in deployment e2e tests, we use `WaitForDeploymentStatus` to wait for deployment rolling update to complete, as well as to make sure that deployments' `maxUnavailable` and `maxSurge` are never violated. This is not covered by unit tests, and most deployment e2e tests aren't fully covered by unit tests.

File: test/e2e/deployment.go [hunk 1]
```diff
@@ -62,45 +59,12 @@ var _ = framework.KubeDescribe("Deployment", func() {
...
 		testScaledRolloutDeployment(f)
 	})
 	It("overlapping deployment should not fight with each other", func() {
 		testOverlappingDeployment(f)
 	})
-	It("lack of progress should be reported in the deployment status", func() {
-		testFailedDeployment(f)
-	})
-	It("iterative rollouts should eventually progress", func() {
-		testIterativeDeployments(f)
```
Comment (kargakis) [inline test/e2e/deployment.go]: Can't remove this

File: test/e2e/deployment.go [hunk 2]
```diff
@@ -62,45 +59,12 @@ var _ = framework.KubeDescribe("Deployment", func() {
...
-	It("RecreateDeployment should delete old pods and create new ones", func() {
-		testRecreateDeployment(f)
-	})
-	It("deployment should delete old replica sets", func() {
-		testDeploymentCleanUpPolicy(f)
-	})
-	It("deployment should support rollover", func() {
-		testRolloverDeployment(f)
-	})
-	It("paused deployment should be ignored by the controller", func() {
```
Comment (kargakis) [inline test/e2e/deployment.go]: Maybe this can be removed

Comment (kargakis) [inline test/e2e/deployment.go]: Actually not

**Comment by kargakis** [comment]
The problem is that even if the controller was fully unit tested most of these e2e tests cannot be replaced because of the nature the controller works (a deployment will never be brought to its desired state in one go).

**Comment by spxtr** [comment]
Thanks for explaining. We should look into getting some sort of framework where you can do that sort of test without  needing an entire cluster.

**Pull request closed without merging**